server/server.go

```diff
@@ -85,6 +109,10 @@ func (s *Server) handler(r *http.Request) http.Handler {
 		return s.Heroku
 	}
 
+	if s.OauthRedirectURL != nil && r.URL.Path =="/oauth/exchange" {
```

Comment:
Is the path consistent/reliable enough to use for checking here? Is there a header or something that indicates the request is part of an oauth exchange? It also might be more efficient to reverse the order of these checks, since presumably we'll usually have the redirect URL set in deployed environments?

Reply:
The incoming request should have a GET verb and the request should contain a `code` parameter.  I'm not sure how to extract the request type from the request though.

Reply:
that seems more fragile than what you're doing here, so if this consistently works, probably best to just leave it. If it's coming from github though, I'm surprised they haven't set some X- custom headers

Reply:
I've reordered and added the check for the code parameter.  While it may have some custom headers as well, if that's not well documented behavior [here](https://docs.github.com/en/developers/apps/authorizing-oauth-apps#2-users-are-redirected-back-to-your-site-by-github) then I'd rather not rely on it, since it could technically change at any time.